packages/plugin-ext/src/plugin/node/debug/debug.ts

```diff
@@ -394,6 +406,9 @@ export class DebugExtImpl implements DebugExt {
         return result;
     }
 
+    /**
+     * @deprecated since 1.25.0. Use $registerDebugConfigurationProvider with $resolveDebugConfigurationByHandle instead.
+     */
     async $resolveDebugConfigurations(debugConfiguration: theia.DebugConfiguration, workspaceFolderUri: string | undefined): Promise<theia.DebugConfiguration | undefined> {
```

Comment:
Given how the method is pretty internal to the plugin-system (and not easily extended), it would be fine to simply break and not deprecate.